dspace-api/src/main/java/org/dspace/app/orcid/client/OrcidConfiguration.java

```diff
@@ -0,0 +1,114 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.orcid.client;
+
+import org.apache.commons.lang3.StringUtils;
+
+/**
+ * A class that contains all the configurations related to ORCID.
+ *
+ * @author Luca Giamminonni (luca.giamminonni at 4science.it)
+ *
+ */
+public final class OrcidConfiguration {
```

Comment:
Since I'm guessing most of these `OrcidConfiguration` properties, configured via `config/spring/api/orcid-services.xml` and set in `config/modules/orcid.cfg`, are required, should these required properties not be constructor parameters, set via spring `<constructor-arg>`, not properties, so it is checked whether all required properties have a (non-null value). Presumably the non-null/not empty check can already be done there if the OrcidAuthentication is configured. PR introduces a `OrcidConfiguration#isApiConfigured` that checks if id & secret aren't blank, but then it's never used?
With comments / documentation in either config files or this class on what each field is / what values expected / how to create/request id/secret.